cluster/cluster.go

```diff
@@ -17,7 +18,8 @@ type Cluster interface {
 	SomeControllers([]flux.ResourceID) ([]Controller, error)
 	Ping() error
 	Export() ([]byte, error)
-	Sync(SyncDef) error
+	ExportByLabel(string, string) ([]byte, error)
+	Sync(SyncDef, map[string]policy.Update, map[string]policy.Update) error
```

Comment:
I would prefer it if the extra metadata were conveyed in the SyncDef's themselves, rather than in parallel in these extra args. This seems a somewhat indirect way of carrying that information.